src/services/codefixes/inferFromUsage.ts

```diff
@@ -265,6 +295,12 @@ namespace ts.codefix {
         addJSDocTags(changes, sourceFile, signature, paramTags);
     }
 
+    function annotateJSDocThis(changes: textChanges.ChangeTracker, sourceFile: SourceFile, containingFunction: FunctionLike, typeNode: TypeNode, _program: Program, _host: LanguageServiceHost) {
+        addJSDocTags(changes, sourceFile, containingFunction, [
+            createJSDocParamTag(createIdentifier("this"), /* isBracketed */ false, createJSDocTypeExpression(typeNode)),

```

Comment:
Can you add a test for this? Specifically, I want to see

1. That the type has braces, like `{Type}`.
2. That it works on a function with existing jsdoc.

Reply:
You know, I haven't actually been able to get this fix to trigger on `.js` files with `JSDoc`. I'll just remove it from this PR.

Reply:
Added. It's unfortunate that instead of copying a `@type {import('someFile.ts').Container}`, it goes with `@type {any}`.